### PR TITLE
Only run e2e on R4R

### PR DESCRIPTION
**Pull request by chatton**
## Description

part of: #3329

This PR makes it so e2es only run when a PR is R4R

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
             done
 
     e2e:
-      if: ${{ !github.event.pull_request.head.repo.fork && github.actor != 'dependabot[bot]' }}
+      if: ${{ !github.event.pull_request.draft && !github.event.pull_request.head.repo.fork && github.actor != 'dependabot[bot]' }}
```
Comment (colin-axner) [inline .github/workflows/e2e.yaml:57]: I think we have to rerun the tests to get this to trigger when marked as r4r?

**Comment by chatton** [comment]
@colin-axner , this combination of types should work. I left comments to be explicit about each one. Let me know what you think.

**Review by colin-axner — APPROVED** [review]
:magic_wand: :heart:

**Review by crodriguezvega — APPROVED** [review]
Thanks for quick improvement, @chatton!

**Pull request merged**